src/hosts.c

```diff
@@ -240,14 +240,20 @@ hosts_read_data (void)
 
   while (h)
     {
+      char *host_deny = NULL;
+
       if (!h->ip)
         {
           /* Scan started. */
           h->ip = kb_item_get_str (h->host_kb, "internal/ip");
           if (h->ip)
             host_set_time (h->host_kb, "internal/start_time");
+          else
+            /* internal/host_deny is set during check_host_authorization() */
+            host_deny = kb_item_get_str (h->host_kb, "internal/host_deny");
         }
-      if (h->ip)
+
+      if (h->ip || host_deny)
```

Comment:
Maybe `if (h->ip ||  !(host->access_allowed))`

See previous comment for context.